tests/run_tests.py

```diff
@@ -316,30 +315,79 @@ def run_system(config_file, vic_exe, test_data_dir, out_dir, driver):
         with open(infile, 'r') as global_file:
             global_param = global_file.read()
 
+        # If restart test, prepare running periods
+        # (1) Find STATESEC option (and STATE_FORMAT option for later use)
+        statesec = find_global_param_value(global_param, 'STATESEC')
+        if driver == 'classic':
+            state_format = find_global_param_value(global_param,
+                                                   'STATE_FORMAT')
+        # (2) Prepare running periods and initial state file info for restart
+        # test
+        if 'restart' in test_dict:
+            run_periods = prepare_restart_run_periods(
+                                test_dict['restart'],
+                                dirs['state'], statesec)
+
         # create template string
         s = string.Template(global_param)
 
         # fill in global parameter options
-        global_param = s.safe_substitute(test_data_dir=test_data_dir,
-                                         result_dir=dirs['results'],
-                                         state_dir=dirs['state'],
-                                         testname=testname,
-                                         test_root=test_dir)
+        # --- if restart test, multiple runs --- #
+        if 'restart' in test_dict:
+            # Set up subdirectories and fill in global parameter options
+            # for restart testing
+            list_global_param =\
+                setup_subdirs_and_fill_in_global_param_restart_test(
+                    s, run_periods, driver, dirs['results'], dirs['state'],
+                    test_data_dir)
+        # --- Else, single run --- #
+        else:
+            global_param = s.safe_substitute(test_data_dir=test_data_dir,
+                                             result_dir=dirs['results'],
+                                             state_dir=dirs['state'])
 
         # replace global options from config file
+        # --- extract global options to be substitute --- #
         if 'options' in test_dict:
             replacements = test_dict['options']
         else:
             replacements = OrderedDict()
-        global_param = replace_global_values(global_param, replacements)
+        # --- if STATE_FORMAT is specified, then the specified value (instead
+        # of the one in the global template file) --- #
+        if 'STATE_FORMAT' in replacements:
+            state_format = replacements['STATE_FORMAT']
+        # --- replace global options --- #
+        if 'restart' in test_dict:
```

Comment:
rather than checking for the `restart` dict, I think it makes more sense to check for the check type `exact_restart`.